modules/tinymce/src/plugins/advlist/main/ts/ui/Buttons.ts

```diff
@@ -80,21 +85,22 @@ const addSplitButton = (editor: Editor, id: string, tooltip: string, cmd: string
   });
 };
 
-const addButton = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, _styles: string[]): void => {
+const addButton = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, styleValue: string): void => {
   editor.ui.registry.addToggleButton(id, {
     active: false,
     tooltip,
     icon: nodeName === ListType.OrderedList ? 'ordered-list' : 'unordered-list',
     onSetup: makeSetupHandler(editor, nodeName),
-    onAction: () => editor.execCommand(cmd)
+    // Need to make sure the button removes rather than applies if a list of the same type is selected
+    onAction: () => editor.queryCommandState(cmd) || styleValue === '' ? editor.execCommand(cmd) : Actions.applyListFormat(editor, nodeName, styleValue)
   });
 };
 
 const addControl = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, styles: string[]): void => {
   if (styles.length > 1) {
     addSplitButton(editor, id, tooltip, cmd, nodeName, styles);
   } else {
-    addButton(editor, id, tooltip, cmd, nodeName, styles);
+    addButton(editor, id, tooltip, cmd, nodeName, normalizeStyleValue(styles[0]));
```

Comment:
I wonder if we should just be normalizing these originally? That'd also save normalizing them every time the split toolbar opens. It also makes sure the logic is consistent, as I noticed this doesn't hand the `disc` or `decimal` styles being transformed to `default` like they do in the split buttons. So something like this as an example:
```ts
const normalizeStyleValue = (styleValue: string): string => {
  const style = styleValue === 'disc' || styleValue === 'decimal' ? 'default' : styleValue;
  return style === 'default' ? '' : style;
}

const addControl = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, styles: string[]): void => {
  const normalizedValues = Arr.map(styles, normalizeStyleValue);
  if (styles.length > 1) {
    addSplitButton(editor, id, tooltip, cmd, nodeName, normalizedValues);
  } else {
    addButton(editor, id, tooltip, cmd, nodeName, normalizedValues[0] ?? '');
  }
};
```